lib/Interpreter/Transaction.cpp

```diff
@@ -67,11 +67,12 @@ namespace cling {
     for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
         if (DI->getKind() == Decl::LinkageSpec) {
-          for ( Decl *DI : LinkageSpecDecl::castToDeclContext(
-                static_cast<LinkageSpecDecl*>(DI))->decls()) {
-            if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
-              if (name.equals(ND->getNameAsString()))
-                return ND;
+          if (LinkageSpecDecl* LSD = cast<LinkageSpecDecl>(DI)) {
```

Comment:
We've already checked DI->getKind() == Decl::LinkageSpec, proper usage according to [http://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates](url)